test/functional/searches_controller_test.rb

```diff
@@ -123,6 +123,27 @@ class SearchesControllerTest < ActionController::TestCase
     end
   end
 
+  context "on GET to show with search parameters with yanked gems" do
+    setup do
+      @sinatra = create(:rubygem, name: "sinatra")
+      @sinatra_redux = create(:rubygem, name: "sinatra-redux")
+      create(:version, rubygem: @sinatra)
+      create(:version, :yanked, rubygem: @sinatra_redux)
+      import_and_refresh
+      @request.cookies["new_search"] = "true"
+      get :show, params: { query: @sinatra_redux.name.to_s }
```

Comment:
please updated this to use `yanked` param.
`@request.cookies["new_search"] = "true"` is not needed. remove it from your setup and maybe send a new PR updating old these where this is still being used.